save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityTagsComponent.kt

```diff
@@ -0,0 +1,127 @@
+@file:Suppress("FILE_NAME_MATCH_CLASS")
+
+package com.saveourtool.save.frontend.components.views.vuln
+
+import com.saveourtool.save.frontend.externals.fontawesome.faPlus
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.frontend.utils.noopResponseHandler
+
+import js.core.jso
+import react.FC
+import react.Props
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.input
+import react.useState
+import web.cssom.ClassName
+
+import kotlinx.browser.window
+
+/**
+ * Component responsible for tag displaying and modifying
+ */
+val vulnerabilityTagsComponent: FC<VulnerabilityTagsComponentProps> = FC { props ->
+    val (newTag, setNewTag) = useState("")
+    val (tagToDelete, setTagToDelete) = useState<String?>(null)
+    val (tagToAdd, setTagToAdd) = useState<String?>(null)
+
+    useRequest(arrayOf(tagToDelete)) {
+        if (!tagToDelete.isNullOrBlank()) {
+            val response = delete(
+                url = "$apiUrl/vulnerabilities/delete-tag",
+                headers = jsonHeaders,
+                params = jso<dynamic> {
+                    vulnerabilityName = props.vulnerabilityName
+                    tag = tagToDelete
+                },
+                loadingHandler = ::loadingHandler,
+                responseHandler = ::noopResponseHandler,
+            )
+            if (response.ok) {
+                props.fetchVulnerability()
+            } else {
+                window.alert(response.unpackMessageOrHttpStatus())
+            }
+            setTagToDelete(null)
+        }
+    }
+
+    useRequest(arrayOf(tagToAdd)) {
+        if (!tagToAdd.isNullOrBlank()) {
+            val response = post(
+                url = "$apiUrl/vulnerabilities/save-tag",
+                headers = jsonHeaders,
+                params = jso<dynamic> {
+                    vulnerabilityName = props.vulnerabilityName
+                    tag = tagToAdd
+                },
+                body = undefined,
+                loadingHandler = ::loadingHandler,
+                responseHandler = ::noopResponseHandler,
+            )
+            if (response.ok) {
+                props.fetchVulnerability()
+            } else {
+                window.alert(response.unpackMessageOrHttpStatus())
+            }
+            setTagToAdd(null)
+        }
+    }
+
+    if (props.canEditVulnerability) {
+        div {
+            className = ClassName("input-group")
+            input {
```

Comment:
may be force tags to be from 5..15 symbols, for example

Reply:
Done